packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -537,29 +556,41 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   static RoutePredicate withName(String name) {
     return (Route<dynamic> route) {
       return !route.willHandlePopInternally
-        && route is ModalRoute && route.settings.name == name;
+          && route is ModalRoute
+          && route.settings.name == name;
     };
   }
 
   // The API for subclasses to override - used by _ModalScope
 
   /// Override this method to build the primary content of this route.
   ///
-  /// * [context] The context in which the route is being built.
-  /// * [animation] The animation for this route's transition. When entering,
-  ///   the animation runs forward from 0.0 to 1.0. When exiting, this animation
-  ///   runs backwards from 1.0 to 0.0.
-  /// * [secondaryAnimation] The animation for the route being pushed on top of
-  ///   this route. This animation lets this route coordinate with the entrance
-  ///   and exit transition of routes pushed on top of this route.
+  /// The arguments have the following meanings:
+  ///
+  ///  * `context`: The context in which the route is being built.
+  ///  * [animation]: The animation for this route's transition. When entering,
+  ///    the animation runs forward from 0.0 to 1.0. When exiting, this animation
+  ///    runs backwards from 1.0 to 0.0.
+  ///  * [secondaryAnimation]: The animation for the route being pushed on top of
+  ///    this route. This animation lets this route coordinate with the entrance
+  ///    and exit transition of routes pushed on top of this route.
+  ///
+  /// This method is called when the route is first built, and rarely
```

Comment:
Not entirely clear what we're proposing to the user. Doesn't sound like we're asking them to build the page content here. Perhaps an example of what might the user want to build each time the route state changes?